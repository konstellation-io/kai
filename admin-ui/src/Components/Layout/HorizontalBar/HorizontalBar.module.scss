@import 'Styles/colors';

.container {
  width: 100%;
  display: flex;
  flex-direction: row-reverse;
  background-color: $bg-color-dark;
}
