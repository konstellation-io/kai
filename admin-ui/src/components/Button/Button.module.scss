@import '../../styles/colors';
@import '../../styles/mixins';
@import '../../styles/variables';

.btn {
  @include font-button;
  position: relative;
  text-align: center;
  padding: 0 42px;
  white-space: nowrap;
  cursor: pointer;
  transition: color ease 0.4s, background-color ease 0.4s;

  display: flex;
  flex-direction: row;
  align-items: center;

  span {
    width: 100%;
  }

  svg {
    margin-right: 14%;
  }

  span {
    width: 100%;
  }

  &.middle {
    text-align: middle;
    padding: 0 20px;
  }
  &.left {
    text-align: left;
    padding: 0 20px;
  }
  &.right {
    text-align: right;
    padding: 0 20px;
  }

  &:hover {
    color: $font-color-white;
  }

  &:active {
    color: $font-color;
  }

  &.border {
    border: solid 1px $color-blue;
  }
}

.primary {
  color: #1a1b1f;
  background-color: $color-button-blue;

  &:active {
    box-shadow: 0 0 20px 0 $color-button-blue;
  }
}

.dark {
  color: $font-color;
  background-color: black;

  &:hover {
    color: $font-color-white;
  }

  &:active {
    box-shadow: none;
  }
}

.grey {
  color: $font-color;
  background-color: $color-grey-darker;

  &:hover {
    color: $font-color-white;
  }

  &:active {
    box-shadow: none;
  }
}

.transparent {
  color: $font-color;
  background-color: transparent;

  &:hover {
    color: $font-color-white;
  }

  &:active {
    box-shadow: none;
  }

  &.primary {
    background-color: rgba(191, 207, 255, 0.15);
    color: white;
  }
}

.label {
  color: $font-color;
}

.disabled {
  color: $font-color;
  pointer-events: none;
  background-color: $color-grey;
}
