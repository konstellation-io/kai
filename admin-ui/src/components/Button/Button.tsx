import React from 'react';
import { Link } from 'react-router-dom';

import Spinner from '../../components/Spinner/Spinner';

import cx from 'classnames';
import styles from './Button.module.scss';

export const BUTTON_TYPES = {
  DEFAULT: 'default',
  DARK: 'dark',
  GREY: 'grey',
  TRANSPARENT: 'transparent'
};
export const BUTTON_ALIGN = {
  LEFT: 'left',
  MIDDLE: 'middle',
  RIGHT: 'right'
};

type Props = {
  label?: string;
  type?: string;
  border?: boolean;
  Icon?: any;
  to?: string;
  onClick?: any;
  primary?: boolean;
  disabled?: boolean;
  loading?: boolean;
  height?: number;
  align?: string;
  style?: Object;
};

function Button({
  type = BUTTON_TYPES.DEFAULT,
  border = false,
  label = 'Button',
  Icon = undefined,
  to = '',
  onClick = function() {},
  primary = false,
  disabled = false,
  loading = false,
  height = 40,
  align = BUTTON_ALIGN.MIDDLE,
  style = {}
}: Props) {
  const content = loading ? (
    <Spinner size={30} color="#0d0e11" />
  ) : (
    <>
      {Icon && <Icon style={{ fontSize: '1rem' }} />}
      <span>{label}</span>
    </>
  );

  const btn = (
    <div
      className={cx(styles.btn, styles[type], styles[align], {
        [styles.primary]: primary,
        [styles.border]: border,
        [styles.label]: !primary,
        [styles.disabled]: disabled
      })}
      style={{ ...style, height, lineHeight: `${height}px` }}
      onClick={onClick}
    >
      {content}
    </div>
  );

  return !disabled && to !== '' ? (
    <Link to={to} data-testid="buttonLink">
      {btn}
    </Link>
  ) : (
    btn
  );
}

export default Button;
