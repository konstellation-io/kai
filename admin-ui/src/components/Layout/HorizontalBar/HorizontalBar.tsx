import React, { FunctionComponent } from 'react';
import styles from './HorizontalBar.module.scss';
import cx from 'classnames';

type Props = {
  children: any;
  style?: string;
};

const HorizontalBar: FunctionComponent<Props> = ({ children, style }) => (
  <div className={cx(styles.container, style)}>{children}</div>
);

export default HorizontalBar;
