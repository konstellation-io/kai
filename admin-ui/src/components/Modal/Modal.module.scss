@import '../../styles/colors';
@import '../../styles/mixins';

.container {
  display: none;
  padding: 14px 0 0 21px;
  background-color: $bg-color-dark;
  max-width: 278px;

  position: absolute;
  top: 25%;
  left: 50%;
  transform: translate(-50%, -25%);

  &.visible {
    display: block;
  }
}

.title {
  @include font-caption;
  color: white;
}

.message {
  @include font-body;
  font-weight: 500;

  margin: 20px 21px 65px 0;
}
