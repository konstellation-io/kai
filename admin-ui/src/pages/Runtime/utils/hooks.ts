import { useMutation } from '@apollo/react-hooks';
import {
  ACTIVATE_VERSION,
  DEACTIVATE_VERSION,
  DEPLOY_VERSION,
  STOP_VERSION,
  ActivateVersionResponse,
  DeactivateVersionResponse,
  DeployVersionResponse,
  StopVersionResponse,
  VersionActionVars
} from '../pages/RuntimeStatusPreview/RuntimeStatusPreview.graphql';

export enum versionActions {
  activate = 'activateVersion',
  deploy = 'deployVersion',
  stop = 'stopVersion',
  deactivate = 'deactivateVersion'
}

export default function useVersionAction() {
  const [activateMutation, { loading: loadingM1 }] = useMutation<
    ActivateVersionResponse,
    VersionActionVars
  >(ACTIVATE_VERSION);
  const [deactivateMutation, { loading: loadingM2 }] = useMutation<
    DeactivateVersionResponse,
    VersionActionVars
  >(DEACTIVATE_VERSION);
  const [deployMutation, { loading: loadingM3 }] = useMutation<
    DeployVersionResponse,
    VersionActionVars
  >(DEPLOY_VERSION);
  const [stopMutation, { loading: loadingM4 }] = useMutation<
    StopVersionResponse,
    VersionActionVars
  >(STOP_VERSION);

  const mutationLoading = [loadingM1, loadingM2, loadingM3, loadingM4].some(
    el => el
  );

  function getMutationVars(versionId: string, comment?: string) {
    const variables = {
      variables: {
        input: {
          versionId: versionId
        }
      }
    } as any;

    if (comment) {
      variables.variables.input.comment = comment;
    }

    return variables;
  }

  return {
    [versionActions.activate]: activateMutation,
    [versionActions.deploy]: deployMutation,
    [versionActions.stop]: stopMutation,
    [versionActions.deactivate]: deactivateMutation,
    getMutationVars,
    mutationLoading
  };
}
