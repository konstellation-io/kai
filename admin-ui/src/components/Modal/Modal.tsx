import React, { useState, MouseEvent } from 'react';

import HorizontalBar from '../Layout/HorizontalBar/HorizontalBar';
import Button from '../Button/Button';

import cx from 'classnames';
import styles from './Modal.module.scss';

type Props = {
  title: string;
  message: string;
  actionButtonLabel: string;
  to?: string;
  blocking?: boolean;
  onAccept?: (e: MouseEvent<HTMLDivElement>) => void;
  onClose?: Function | null;
};

function Modal({
  title,
  message,
  actionButtonLabel,
  to = '',
  blocking = false,
  onAccept = function() {},
  onClose = null
}: Props) {
  const [isVisible, setIsVisible] = useState(true);

  function onCancelClick() {
    if (onClose) onClose();
    else setIsVisible(false);
  }

  return (
    <>
      {blocking && <div className={styles.bg} />}
      <div
        className={cx(styles.container, {
          [styles.visible]: isVisible
        })}
        data-testid="modal-container"
      >
        <div className={styles.title} data-testid="modal-title">
          {title}
        </div>
        <div className={styles.message} data-testid="modal-message">
          {message}
        </div>
        <HorizontalBar>
          <Button
            primary
            label={actionButtonLabel}
            to={to}
            onClick={onAccept}
            height={30}
            style={{ width: '122px', padding: '0 0' }}
          />
          <Button
            label={'CANCEL'}
            onClick={onCancelClick}
            height={30}
            style={{ width: '122px', padding: '0 0' }}
          />
        </HorizontalBar>
      </div>
    </>
  );
}

export default Modal;
