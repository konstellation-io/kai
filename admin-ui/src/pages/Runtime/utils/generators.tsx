import React from 'react';
import Button from '../../../components/Button/Button';
import StartIcon from '@material-ui/icons/SkipNext';
import StopIcon from '@material-ui/icons/PauseCircleFilled';
import PublishIcon from '@material-ui/icons/PlayCircleFilledOutlined';
import UnpublishIcon from '@material-ui/icons/Block';
import { VersionStatus } from '../../../graphql/types/globalTypes';

function generateActionButton(label: string, icon: any, action: Function) {
  return (
    <Button
      key={label}
      label={label}
      onClick={action}
      height={30}
      Icon={icon}
    />
  );
}

function getStateToButtons(
  publishAction: Function,
  startAction: Function,
  stopAction: Function,
  unpublishAction: Function
) {
  const buttonStart = generateActionButton('START', StartIcon, startAction);
  const buttonStop = generateActionButton('STOP', StopIcon, stopAction);
  const buttonPublish = generateActionButton(
    'PUBLISH',
    PublishIcon,
    publishAction
  );
  const buttonUnpublish = generateActionButton(
    'UNPUBLISH',
    UnpublishIcon,
    unpublishAction
  );

  return {
    [VersionStatus.STOPPED]: [buttonStart],
    [VersionStatus.PUBLISHED]: [buttonUnpublish],
    [VersionStatus.STARTED]: [buttonPublish, buttonStop],
    [VersionStatus.STOPPED]: [buttonStart]
  };
}

export function getVersionActionButtons(
  publishAction: Function,
  startAction: Function,
  stopAction: Function,
  unpublishAction: Function,
  status?: string
) {
  const stateToButtons: { [key: string]: any } = getStateToButtons(
    publishAction,
    startAction,
    stopAction,
    unpublishAction
  );

  return stateToButtons[status || ''];
}
