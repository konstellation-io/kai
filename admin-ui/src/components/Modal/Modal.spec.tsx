import React from 'react';
import Modal from './Modal';
import { testid } from '../../utils/testUtilsEnzyme';
import HorizontalBar from '../Layout/HorizontalBar/HorizontalBar';
import Button from '../Button/Button';
import { shallow } from 'enzyme';

const ACTION_BUTTON_LABEL = 'Action Button';

const mockClick = jest.fn();

describe('Modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Modal
        title="Modal"
        message="Message"
        actionButtonLabel={ACTION_BUTTON_LABEL}
        onAccept={mockClick}
      />
    );
  });

  it('works with default props', () => {
    expect(wrapper.exists(HorizontalBar)).toBeTruthy();
    expect(wrapper.find('.title').text()).toBe('Modal');
    expect(wrapper.find('.message').text()).toBe('Message');
    expect(
      wrapper.find({ label: 'Action Button' }).exists(Button)
    ).toBeTruthy();
  });

  it('hides when clicking on Cancel', () => {
    wrapper.find({ label: 'CANCEL' }).simulate('click');

    expect(
      wrapper.find(testid('modal-container')).exists('.visible')
    ).toBeFalsy();
  });

  it('shows shield when blocking', () => {
    wrapper.setProps({ blocking: true });

    expect(wrapper.exists('.bg')).toBeTruthy();
  });

  it('handles click event on Accept click', () => {
    wrapper.find({ label: 'Action Button' }).simulate('click');

    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});
