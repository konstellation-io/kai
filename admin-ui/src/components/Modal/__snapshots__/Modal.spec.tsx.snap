// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<Fragment>
  <div
    className="container visible"
    data-testid="modal-container"
  >
    <div
      className="title"
      data-testid="modal-title"
    >
      Modal
    </div>
    <div
      className="message"
      data-testid="modal-message"
    >
      Message
    </div>
    <HorizontalBar>
      <Button
        height={30}
        label="Action Button"
        onClick={[MockFunction]}
        primary={true}
        style={
          Object {
            "padding": "0 0",
            "width": "122px",
          }
        }
        to=""
      />
      <Button
        height={30}
        label="CANCEL"
        onClick={[Function]}
        style={
          Object {
            "padding": "0 0",
            "width": "122px",
          }
        }
      />
    </HorizontalBar>
  </div>
</Fragment>
`;
